### ISSUE TITLE
Cannot read properties of null (reading 'getTypeChecker')

### ISSUE THREAD

**Issue by Amorim33** [yukukotani/eslint-plugin-chakra-ui#161]
## Error
```
yarn run v1.22.19
$ yarn workspace @project/partners lint
$ eslint --max-warnings=0 "**/*.{ts*,js}"

Oops! Something went wrong! :(

ESLint: 8.46.0

TypeError: Cannot read properties of null (reading 'getTypeChecker')
Occurred while linting /project/packages/partners/src/App.tsx:22
Rule: "chakra-ui/props-order"
    at isChakraElement (/project/node_modules/eslint-plugin-chakra-ui/dist/lib/isChakraElement.js:13:48)
    at JSXOpeningElement (/project/node_modules/eslint-plugin-chakra-ui/dist/rules/props-order.js:76:155)
    at ruleErrorHandler (/project/node_modules/eslint/lib/linter/linter.js:1050:28)
    at /project/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (/project/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (/project/node_modules/eslint/lib/linter/node-event-generator.js:297:26)
    at NodeEventGenerator.applySelectors (/project/node_modules/eslint/lib/linter/node-event-generator.js:326:22)
    at NodeEventGenerator.enterNode (/project/node_modules/eslint/lib/linter/node-event-generator.js:340:14)
    at CodePathAnalyzer.enterNode (/project/node_modules/eslint/lib/linter/code-path-analysis/code-path-analyzer.js:795:23)
error Command failed with exit code 2.
```

It happens because `program` is null.
https://github.com/yukukotani/eslint-plugin-chakra-ui/blob/f84ca014e2310738e3adc80414aa54cea305dc47/src/lib/isChakraElement.ts#L5

## Context
- Monorepo
- TS version: 5.1.6
- ESLINT version: 8.46.0

## Config
 ```
 extends: [
    'eslint:recommended',
    // 'standard-with-typescript',
    'prettier',
    'plugin:@typescript-eslint/recommended',
    'plugin:import/recommended',
    'plugin:import/typescript',
    'plugin:react/recommended',
    'plugin:react/jsx-runtime',
    'plugin:react-hooks/recommended',
    'plugin:jsx-a11y/recommended',
    'plugin:storybook/recommended',
  ],
  parser: '@typescript-eslint/parser',
  plugins: ['@typescript-eslint', 'chakra-ui', 'relay'],
  env: { browser: true, node: true }, // @TODO: Change this based in the package
  parserOptions: {
    ecmaVersion: 'latest',
    sourceType: 'module',
    ecmaFeatures: { jsx: true },
  },
  rules: {
     // chakra-ui rules
    'chakra-ui/props-order': 'error',
    'chakra-ui/props-shorthand': 'error',
    'chakra-ui/require-specific-component': 'error',
    ...
   },
  settings: {
    react: { version: 'detect' },
    'import/resolver': { typescript: true, node: true },
  },
 ```

**Comment by yukukotani** [comment]
@Amorim33 Thank you for the reporting. Which version of `@typescript-eslint/parser` have you installed?

**Comment by hndrr** [comment]
@yukukotani 
I'm also suffering from the same error.

I have tried `@typescript-eslint/parser` versions 6.3.0 and 6.4.1.
I am also running in an environment using vite 4.4.9 and yarn 3.6.1.

```
hndr@hndrnoMacBook-Pro HHD-frontend % yarn lint

Oops! Something went wrong! :(

ESLint: 8.46.0

TypeError: Cannot read properties of null (reading 'getTypeChecker')
Occurred while linting /Users/hndr/workspace/HHD-frontend/src/App.tsx:7
Rule: "chakra-ui/props-order"
    at isChakraElement (/Users/hndr/workspace/HHD-frontend/node_modules/eslint-plugin-chakra-ui/dist/lib/isChakraElement.js:13:48)
    at JSXOpeningElement (/Users/hndr/workspace/HHD-frontend/node_modules/eslint-plugin-chakra-ui/dist/rules/props-order.js:76:155)
    at ruleErrorHandler (/Users/hndr/workspace/HHD-frontend/node_modules/eslint/lib/linter/linter.js:1050:28)
    at /Users/hndr/workspace/HHD-frontend/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (/Users/hndr/workspace/HHD-frontend/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (/Users/hndr/workspace/HHD-frontend/node_modules/eslint/lib/linter/node-event-generator.js:297:26)
    at NodeEventGenerator.applySelectors (/Users/hndr/workspace/HHD-frontend/node_modules/eslint/lib/linter/node-event-generator.js:326:22)
    at NodeEventGenerator.enterNode (/Users/hndr/workspace/HHD-frontend/node_modules/eslint/lib/linter/node-event-generator.js:340:14)
    at CodePathAnalyzer.enterNode (/Users/hndr/workspace/HHD-frontend/node_modules/eslint/lib/linter/code-path-analysis/code-path-analyzer.js:795:23)
hndr@hndrnoMacBook-Pro HHD-frontend % 
```

**Comment by yukukotani** [comment]
@Amorim33 @hndrr Fixed in v10.0.0 by #163

**Comment by hndrr** [comment]
@yukukotani 
I checked the operation and it seemed to work fine.
Thank you very much.
